src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -249,6 +258,10 @@ public BaseResponseBean advanceSessionWithSelections(MenuSession menuSession,
                     return syncResponse;
                 }
             }
+
+            if (nextScreen == null) {
+                executeAndRebuildSession(menuSession);
```

Comment:
this is necessary to execute any pending stack ops steps caused by a case claim mark/rewind mechanism. Earlier this step would only happen outside of this `selections` loop, though to support parent-child navigation with case claim, we need to resolve pending stack ops in the current `selections` loop itself. For eg, If we have a parent child navigation setup and user selected a parent and then a child menu, the selections would look something like [parent_menu, parent_case_id, child_menu]. Now while replaying the session with auto launch set on the parent case list, on processing `parent_case_id` the session state would look something like this - 

```

COMMAND_ID m1
MARK case_id
COMMAND_ID search_command.m1
QUERY_REQUEST results : http://localhost:8000/a/test-1/phone/search/)
CASE_ID parent_case_id
```
At this state, needed datum will be null (hence the next screen) until we perform the rewind operation.

Reply:
I think this is the key I was missing when trying to implement this previously.